src/sse/sse_radix4.rs

```diff
@@ -0,0 +1,518 @@
+
+        // Base-level FFTs
+        match &self.base_fft {
+            Sse32Butterfly::Len1(bf) => bf.perform_fft_butterfly_multi(spectrum).unwrap(),
+            Sse32Butterfly::Len2(bf) => bf.perform_fft_butterfly_multi(spectrum).unwrap(),
+            Sse32Butterfly::Len4(bf) => bf.perform_fft_butterfly_multi(spectrum).unwrap(),
+            Sse32Butterfly::Len8(bf) => bf.perform_fft_butterfly_multi(spectrum).unwrap(),
+            Sse32Butterfly::Len16(bf) => bf.perform_fft_butterfly_multi(spectrum).unwrap(),
+            Sse32Butterfly::Len32(bf) => bf.perform_fft_butterfly_multi(spectrum).unwrap(),
+        };
+
+        // cross-FFTs
+        let mut current_size = self.base_len * 4;
+        let mut layer_twiddles: &[__m128] = &self.twiddles;
+
+        while current_size <= signal.len() {
+            let num_rows = signal.len() / current_size;
+
+            for i in 0..num_rows {
+                butterfly_4_32(
+                    &mut spectrum[i * current_size..],
+                    layer_twiddles,
+                    current_size / 4,
+                    &self.bf4,
+                )
+            }
+
+            //skip past all the twiddle factors used in this layer
+            let twiddle_offset = (current_size * 3) / 8;
+            layer_twiddles = &layer_twiddles[twiddle_offset..];
+
+            current_size *= 4;
+        }
+    }
+}
+boilerplate_fft_sse_oop!(Sse32Radix4, |this: &Sse32Radix4<_>| this.len);
+
+#[target_feature(enable = "sse3")]
+unsafe fn butterfly_4_32<T: FftNum>(
+    data: &mut [Complex<T>],
+    twiddles: &[__m128],
+    num_ffts: usize,
+    bf4: &SseF32Butterfly4<T>,
+) {
+    let mut idx = 0usize;
+    let mut tw_idx = 0usize;
+    let output_slice = data.as_mut_ptr() as *mut Complex<f32>;
+    for _ in 0..num_ffts / 4 {
+        let scratch0 = _mm_loadu_ps(output_slice.add(idx) as *const f32);
+        let scratch0b = _mm_loadu_ps(output_slice.add(idx + 2) as *const f32);
+
+        let mut scratch1 = _mm_loadu_ps(output_slice.add(idx + 1 * num_ffts) as *const f32);
+        let mut scratch1b = _mm_loadu_ps(output_slice.add(idx + 2 + 1 * num_ffts) as *const f32);
+
+        let mut scratch2 = _mm_loadu_ps(output_slice.add(idx + 2 * num_ffts) as *const f32);
+        let mut scratch2b = _mm_loadu_ps(output_slice.add(idx + 2 + 2 * num_ffts) as *const f32);
+
+        let mut scratch3 = _mm_loadu_ps(output_slice.add(idx + 3 * num_ffts) as *const f32);
+        let mut scratch3b = _mm_loadu_ps(output_slice.add(idx + 2 + 3 * num_ffts) as *const f32);
+
+        scratch1 = complex_dual_mul_f32(scratch1, twiddles[tw_idx]);
+        scratch2 = complex_dual_mul_f32(scratch2, twiddles[tw_idx + 1]);
+        scratch3 = complex_dual_mul_f32(scratch3, twiddles[tw_idx + 2]);
+        scratch1b = complex_dual_mul_f32(scratch1b, twiddles[tw_idx + 3]);
+        scratch2b = complex_dual_mul_f32(scratch2b, twiddles[tw_idx + 4]);
+        scratch3b = complex_dual_mul_f32(scratch3b, twiddles[tw_idx + 5]);
```

Comment:
Does it improve performance at all if you loop over twiddles with `chunks_exact` here?

Reply:
Will try!

Reply:
Yes that does speed things up a little bit. That should work on the scalar radix4 as well. I'll work a bit on that once I'm done with this one.